pkg/authorization/api/helpers.go

```diff
@@ -38,6 +61,26 @@ func ExpandResources(rawResources sets.String) sets.String {
 	return ret
 }
 
+func needsNormalizing(in string) bool {
+	// check to see if we conform to common cases for case
+	// this looks insane.  Follow me here and blame Jordan
+	// strings can be cast to []byte for free.  We range over the bytes to avoid
+	// the conversion to runes that happens when ranging over strings.  This results in
+	// zero allocations here.
+	for _, currByte := range []byte(in) {
```

Comment:
> I started weaving my way through string ranging and unicode.IsLower() to convince myself that it was cheap for ascii strings... a isLowerString(s string) would make me happier, but I hadn't proven it to myself yet

Given the pain of analysis, is it worth it?  We all wrote this once upon a time in school, so it may look stupid, but it's easy to read, it's never a false negative, and it doesn't require delving into the depths of runes (that no one else will ever read).